pages/plants/[id].js

```diff
@@ -0,0 +1,116 @@
+import { useRouter } from "next/router";
+import plants from "@/lib/data";
+import Image from "next/image";
+import { IoArrowBackOutline } from "react-icons/io5";
+import styled from "styled-components";
+import { LuDroplet } from "react-icons/lu";
+import { FiSun } from "react-icons/fi";
+
+export default function PlantDetailsPage() {
+  const router = useRouter();
+
+  const { id } = router.query;
+
+  const plant = plants.find((plant) => plant.id === id);
+
+  if (!plant) {
+    return <p>Plant not found!</p>;
+  }
+
+  return (
+    <>
+      <main>
+        <h1>Plant Details</h1>
```

Comment:
Thinking in components I would recommend to render the `<main>` and the `<h1>` here but then render a PlantDetails component. You can copy and paste the code into that component. This again would separate the concerns: You will have a reusable PlantDetails component you can use everywhere in your app.

Reply:
Many thanks for your feedback, Roland. We've implemented a PlantDetails component as you suggested to seperate the concerns between `[id].js` and `PlantDetails.js`.